### PR TITLE
Add Json marshaling and unmarshaling

### LINKED ISSUE #2: Support for JSON marshaling/unmarshaling
To improve the ergonomics of the library when using it for encoding/decoding JSON, I'd like to suggest implementing the [Marshaler](https://pkg.go.dev/encoding/json#Marshaler) and [Unmarshaler](https://pkg.go.dev/encoding/json#Unmarshaler) interfaces

### PR DESCRIPTION
**Pull request by arieroos**
This should close #2 

I thought it's a good idea. And some contributions to this package will help it's longevity

### CODE REVIEW

File: duration_test.go [hunk 1]
```diff
@@ -262,3 +263,64 @@ func TestDuration_String(t *testing.T) {
...
+	}
+
+	jsonVal, err = json.Marshal(struct {
+		Dur Duration `json:"d"`
+	}{Dur: *td})
+	if err != nil {
+		t.Errorf("did not expect error: %s", err.Error())
+	}
+	if string(jsonVal) != `{"d":"P3Y6M4DT12H30M5.5S"}` {
+		t.Errorf("expected: %s, got: %s", `{"d":"P3Y6M4DT12H30M5.5S"}`, string(jsonVal))
```
Comment (sosodev) [inline duration_test.go:290]: This error is hit.

File: duration.go [hunk 2]
```diff
@@ -275,3 +276,26 @@ func (duration *Duration) String() string {
 
 	return d
 }
+
+func (duration *Duration) MarshalJSON() ([]byte, error) {
```
Comment (sosodev) [inline duration.go]: Remove the pointer`*` here to fix it.

**Review by sosodev — CHANGES_REQUESTED** [review]
Thanks for submitting this. Sorry for the slow review. Overall this looks good but the tests are failing.

File: duration.go (hunk 2 above)
Comment (arieroos) [reply duration.go]: Thanks. I pushed a commit to fix this

**Comment by arieroos** [comment]
Thanks a million for the review. No need to apologise for taking your time.

I pushed a commit with your suggested fix

**Review by sosodev — APPROVED** [review]

**Comment by sosodev** [comment]
@arieroos merged. Thanks again for your contribution. 😄

**Pull request merged**